lib/web_ui/dev/steps/compile_tests_step.dart

```diff
@@ -305,6 +342,50 @@ Future<bool> compileUnitTest(FilePath input, {required Renderer renderer}) async
   }
 }
 
+Future<bool> compileUnitTestToWasm(FilePath input, {required Renderer renderer}) async {
+  final String targetFileName = pathlib.join(
+    environment.webUiBuildDir.path,
+    '${input.relativeToWebUi}.browser_test.dart.wasm',
+  );
+
+  final io.Directory directoryToTarget = io.Directory(pathlib.join(
+      environment.webUiBuildDir.path,
+      pathlib.dirname(input.relativeToWebUi)));
+
+  if (!directoryToTarget.existsSync()) {
+    directoryToTarget.createSync(recursive: true);
+  }
+
+  final List<String> arguments = <String>[
+    environment.dart2wasmSnapshotPath,
+
+    '--dart-sdk=${environment.dartSdkDir.path}',
+
+    // We do not want to auto-select a renderer in tests. As of today, tests
+    // are designed to run in one specific mode. So instead, we specify the
+    // renderer explicitly.
+    '-DFLUTTER_WEB_AUTO_DETECT=false',
+    '-DFLUTTER_WEB_USE_SKIA=${renderer == Renderer.canvasKit}',
```

Comment:
doesn't this codepath always use skwasm? Or are we compiling everything with dart2wasm?